voltixApp/VoltixApp/View Models/SendTransaction.swift

```diff
@@ -37,21 +37,90 @@ class SendTransaction: ObservableObject, Hashable {
     var amountInWei: BigInt {
         BigInt(amountDecimal * pow(10, Double(EVMHelper.ethDecimals)))
     }
-                
+    
+    var isAmountExceeded: Bool {
+        
+        let totalBalance = BigInt(coin.rawBalance) ?? BigInt.zero
+        
+        var gasInt = BigInt.zero
+        if coin.isNativeToken {
+            gasInt = BigInt(gas) ?? BigInt.zero
+            if coin.chainType == .EVM {
+                if let gasLimitBigInt = BigInt(coin.feeDefault) {
+                    gasInt = gasInt * gasLimitBigInt
+                }
+            }
+        }
+        
+        let totalTransactionCost = amountInRaw + gasInt
+        
+        return totalTransactionCost > totalBalance
+        
+    }
+        
+    func hasEnoughNativeTokensToPayTheFees() async -> Bool {
+        guard !coin.isNativeToken else { return true } 
+
+        var gasPriceBigInt = BigInt(gas) ?? BigInt.zero
+        if let gasLimitBigInt = BigInt(coin.feeDefault) {
+            if coin.chainType == .EVM {
+                gasPriceBigInt *= gasLimitBigInt
+            } 
+            if let vault = ApplicationState.shared.currentVault {
```

Comment:
We don't need to do a RPC call to check the balance , balance has been checked in the vault detail view,  so just remove the RPC call , and check the balance field in the coin should be enough.

Reply:
The balance is empty because we get it from the vault store and the vault store has an empty balance. We call the get balance for the native coin only if it has a cache then it will return from the cache.